### PR TITLE
Deploy pool-level SNX rewards distributor

### PR DESCRIPTION
**Pull request by noisekit**
Note that the old one that we used to have is no longer relevant and outdated - so I have just replaced the content of the file with new distributor deployment. Old one will be later cleaned away from cannon state.

Extra changes:
- Skip Legacy Market test
- Optimised collateral types task to use pre-compiled json
- Unify setting name with base mainnet config, rename to `spartan_council_pool_id_setting`
- Fixed AllErrors generation to avoid duplicates
- Fix pool config for base sepolia (align with mainnet)
- Fix cannon warning about deprecated option and switch to use `--anvil.port`

### CODE REVIEW

File: e2e/tests/omnibus-mainnet.toml/LegacyMarket.e2e.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const { contractWrite } = require('../../tasks/contractWrite');
 
 const log = require('debug')(`e2e:${require('path').basename(__filename, '.js')}`);
 
-describe(require('path').basename(__filename, '.e2e.js'), function () {
+describe.skip(require('path').basename(__filename, '.e2e.js'), function () {
```
Comment (noisekit) [inline e2e/tests/omnibus-mainnet.toml/LegacyMarket.e2e.js:11]: Skipping the legacy market test for now as test accounts are already migrated

File: omnibus-base-sepolia-andromeda.toml [hunk 2]
```diff
@@ -319,7 +319,7 @@ args = [
 ]
 depends = ['provision.system', 'setting.account_timeout_withdraw']
 
-[setting.sc_pool_id]
+[setting.spartan_council_pool_id_setting]
```
Comment (noisekit) [inline omnibus-base-sepolia-andromeda.toml:322]: renamed setting to match mainnet

**Review by kaleb-keny — APPROVED** [review]

**Comment by kaleb-keny** [comment]
Hey @noisekit i've approved it, but before we send it to mainnet,  do you think it's wortwhile if we test out a reward distribution and claim on the sepolia? I believe we have plenty of e2e tests, but who knows there could be some specificities related to evm that are not covered by our integration tests... Happy to help out too

**Review by kaleb-keny — APPROVED** [review]

**Comment by noisekit** [comment]
Yes I will send it to the Base Sepolia now. Thanks.

**Comment by noisekit** [comment]
Screenshot 2024-12-10 at 01 48 45
Screenshot 2024-12-10 at 01 49 46
Screenshot 2024-12-10 at 01 54 58

Claimed
https://sepolia.basescan.org/tx/0xfb4d711cd63ed14c9a2e39efc21b617bcea05f1b19f4003b011168487c62f39f


Everything looks ok

**Pull request merged**